karapace/protobuf/schema.py

```diff
@@ -1,14 +1,188 @@
 # Ported from square/wire:
 # wire-library/wire-schema/src/commonMain/kotlin/com/squareup/wire/schema/Schema.kt
 # Ported partially for required functionality.
+from karapace.protobuf.enum_element import EnumElement
+from karapace.protobuf.location import Location
+from karapace.protobuf.message_element import MessageElement
+from karapace.protobuf.option_element import OptionElement
+from karapace.protobuf.proto_file_element import ProtoFileElement
+from karapace.protobuf.proto_parser import ProtoParser
+from karapace.protobuf.utils import append_documentation, append_indented
 
+import logging
 
+log = logging.getLogger(__name__)
+
+
+def static_init(cls):
+    if getattr(cls, "static_init", None):
+        cls.static_init()
+    return cls
+
+
+def add_slashes(text: str):
+    escape_dict = {
+        r'\a': '\\a',
+        r'\b': '\\b',
+        r'\c': '\\c',
+        r'\f': '\\f',
+        r'\n': '\\n',
+        r'\r': '\\r',
+        r'\t': '\\t',
+        r'\v': '\\v',
+        '\'': "\\'",
+        '\"': '\\\"',
+        '\\': '\\\\'
+    }
+    result: str = ""
+    for char in text:
+        try:
+            result += escape_dict[char]
+        except KeyError:
+            result += char
+    return result
+
+
+def message_element_string(element: MessageElement) -> str:
+    result: list = list()
+    append_documentation(result, element.documentation)
+    result.append(f"message {element.name} {{")
+    if element.reserveds:
+        result.append("\n")
```

Comment:
Suggestion: Only add the content to the list, then at the end join using the line-break `'\n'.join(result)`.

Reply:
This code must return exactly the same output as the appropriate SchemaRegistry code. And result array elements contain not only lines but also strings which must be concatenated without a line break. 
This code is part of ported square/wire library where java StringBuilder was used. 
Probably we can change it to concatenation with line break but in this case, we must change a some amount of subroutines to be sure that output is the same as in ShemaRegistry.

Reply:
understood, thanks